# Can you identify myocardial infarction?

In [24]:
import pandas as pd
import numpy as np

In [2]:
import os

In [12]:
os.chdir("E:\Data_Science\Heartbeat\heartbeat")

|**The PTB Diagnostic ECG Database**||
|--|--|
|Number of Samples: |14552|
|Number of Categories:| 2|
|Sampling Frequency: |125Hz|
|Data Source: |Physionet's PTB Diagnostic Database|

In [26]:
ptb_normal = pd.read_csv('ptbdb_normal.csv', header=None)
ptb_abnormal = pd.read_csv('ptbdb_abnormal.csv')

In [27]:
ptb_normal.head()

0         1         2         3         4         5         6    \
0  1.000000  0.900324  0.358590  0.051459  0.046596  0.126823  0.133306   
1  1.000000  0.794681  0.375387  0.116883  0.000000  0.171923  0.283859   
2  0.909029  0.791482  0.423169  0.186712  0.000000  0.007836  0.063032   
3  1.000000  0.478893  0.056760  0.064176  0.081289  0.072732  0.055619   
4  1.000000  0.867238  0.201360  0.099349  0.141336  0.120934  0.108516   

        7         8         9    ...  178  179  180  181  182  183  184  185  \
0  0.119125  0.110616  0.113047  ...  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.293754  0.325912  0.345083  ...  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
2  0.077002  0.074957  0.077342  ...  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
3  0.048774  0.054478  0.041643  ...  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
4  0.096393  0.093436  0.100828  ...  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

   186  187  
0  0.0  0.0  
1  0.0  0.0  
2  0.0  0.0  
3  0.0  0.0  
4  0.0  0.0  

[5 rows x 188 columns]

In [15]:
print(ptb_normal.shape)
print(ptb_abnormal.shape)

(4045, 188)
(10505, 188)


There are 4045 records labeled as normal and 10505 records labeled as abnormal. 

## Visualization

In [28]:
sfreq = 125
# Calculate the time array
time = np.arange(0, len(ptb_normal)) / sfreq

In [31]:
time

array([0.0000e+00, 8.0000e-03, 1.6000e-02, ..., 3.2344e+01, 3.2352e+01,
       3.2360e+01])

In [ ]:
plt.figure(15,7)
